### DISCUSSION TITLE
What do you think about the "new" Microsoft AOT compiler?

### DISCUSSION THREAD

**Discussion by 0xF6** [CosmosOS/Cosmos#1838] [Ideas]
New package that will be distributed in .NET 6 to support compilation in AOT mode, will also support the NO-RUNTIME option.
Which can also be used to build EFI-like applications.

Proof: 
[Variant efi no runtime](https://github.com/MichalStrehovsky/zerosharp/blob/master/efi-no-runtime/zerosharp.cs)
[runtime lab - aot branch](https://github.com/dotnet/runtimelab/tree/feature/NativeAOT)

**Comment by 0xF6** [comment]
It can also help with ARM-like support 🗿

**Comment by quajak** [comment]
We are looking into it. It brings quite a few advantages but also requires a bigger rewrite.

**Reply by 0xF6** [reply]
time to Cosmos 2.0 😃

**Comment by valentinbreiz** [comment]
Hello @0xF6, I'm really interested in this project, I made an issue for it (https://github.com/CosmosOS/Cosmos/issues/1857). This is just an opening to the discussion but I'm very optimistic
